lib/transaction.js

```diff
@@ -225,12 +219,11 @@ class Transaction extends EventEmitter {
       .then((connection) => {
         connection.__knexTxId = this.txid;
 
-        return (this._previousSibling
-          ? this._previousSibling.catch(() => {})
-          : Promise.resolve()
-        ).then(function() {
-          return connection;
-        });
+        return this._previousSibling
+          .catch(() => {}) // TODO: Investigate this line
+          .then(function() {
+            return connection;
+          });
```

Comment:
This was just a minor code cleanup.  It turns out that `this._previousSibling` is _always_ defined due to the way the constructor is written.  So, there was no need for conditional logic here.

Additionally, take note of the `TODO`.  I'm 95% sure that the code should _not_ be ignoring that exception.  However, I left it in place for now so that we can minimize the impact of this PR.

Reply:
Actually... I suspect that the `.catch(()=> {})` line was the reason for the "silent failures" mentioned here:

https://github.com/knex/knex/pull/3328#issuecomment-508403321

I'm going to test out a different implementation that will (hopefully!) allow for proper sibling->sibling linking without swallowing exceptions.

Reply:
Alright -- I just pushed the code for establishing 2 separate Promise Chains.  So, instead of doing this:

```
this._promise = createPromise()
outer._lastChild = this._promise.catch(()=> {})
// Oops!  Now `this.promise` will always think that its exceptions have been handled!
```

It is doing this:

```
const basePromise = createPromise();
this._promise = basePromise.then((x)=> x)
outer._lastChild = this._promise.catch(()=> {})
// Yay!  Now `this._promise` understands that its exceptions still need to be handled!
```